README.md

```diff
@@ -20,20 +20,78 @@ historical purposes, but are not necessarily indicative of the
+standard library will not be considered in this stage.
+
+Stage 2 will commence once the implementation work on the Stage 1
+features is cresting, and can contain a few other large and small
+features. We expect that stage 2 will commence some time in Spring
+2017.
+
+The high-priority features supporting stage 1's source and ABI
+stability goals are:
+
+* Source stability features: the Swift language will need [some
+  accommodations](https://github.com/apple/swift-evolution/blob/master/proposals/0141-available-by-swift-version.md)
+  to support code bases that target different language versions, to
+  help Swift deliver on its source-compatibility goals while still
+  enabling rapid progress.
+
+* Resilience: resilience provides a way for public APIs to evolve over
+  time, while maintaining a stable ABI. For example, resilience
+  eliminates the [fragile base class
+  problem](https://en.wikipedia.org/wiki/Fragile_base_class) that
+  occurs in some object-oriented languages (e.g., C++) by describing
+  the types of API changes that can be made without breaking ABI
+  (e.g., "a new stored property or method can be added to a class").
+
+* Stabilizing the ABI: There are a ton of small details that need to
+  be audited and improved in the code generation model, including
+  interaction with the Swift runtime. While not specifically
+  user-facing, the decisions here affect performance and (in some rare
+  cases) the future evolution of Swift.
+
+* Generics improvements needed by the standard library: the standard
+  library has a number of workarounds for language deficiencies, many
+  of which manifest as extraneous underscored protocols and
+  workarounds. If the underlying language deficiencies remain, they
+  become a permanent part of the stable ABI. [Conditional
+  conformances](https://github.com/apple/swift-evolution/blob/master/proposals/0143-conditional-conformances.md),
+  [recursive protocol
+  requirements](https://github.com/apple/swift/blob/master/docs/GenericsManifesto.md#recursive-protocol-constraints-),
+  and [where clauses for associated
+  types](https://github.com/apple/swift-evolution/blob/master/proposals/0142-associated-types-constraints.md)
+  are known to be in this category, but it's plausible that other
+  features will be in scope if they would be used in the standard
+  library.
+
+* String re-evaluation: String is one of the most important
+  fundamental types in the language. Swift 4 seeks to make strings more
+  powerful and easier-to-use, while retaining Unicode correctness by
+  default.
+
+* Memory ownership model: an (opt-in) Cyclone/Rust-inspired memory
+  ownership model is highly desired by systems programmers and for
+  other high-performance applications that want predictible and
+  deterministic performance. This feature will fundamentally shape the
+  ABI, from low-level language concerns such as "inout" and low-level
+  "addressors" to its impact on the standard library.
```

Comment:
I'd suggest mentioning that we need to get a design of this, which will allow us to scope the small part that effects the ABI and make sure that happens in Stage 1.